crates/ide_completion/src/completions/fn_param.rs

```diff
@@ -143,4 +159,25 @@ fn foo2($0) {}
             "#]],
         )
     }
+
+    #[test]
+    fn test_param_completion_self_param() {
+        check(
+            r#"
+                impl A {
+                    fn foo(file_id: FileId) {}
+                    fn bar(file_id: FileId) {}
+                    fn new($0) {
+                    }
+                }
+            "#,
+            expect![[r#"
+                bn self
+                bn &self
+                bn mut self
+                bn &mut self
+                bn file_id: FileId
```

Comment:
Wait, why are we suggesting `file_id` here?

Reply:
That's our magic param completions, because those parameters appear in other functions in the same file we show them as well.

Reply:
Ooh, that's.. unexpected.

But then `self` and friends should have `CompletionKind::Keyword`, right?

Reply:
Ah, no, that's different from `CompletionItemKind`.

Reply:
`CompletionItemKind` exists solely for testing purposes, I think `Snippet` would probably fit better for it but it doesn't really matter that much as its only used for filtering in tests(and ideally shouldn't exist at all imo which I'm slowly getting at)

Reply:
So just getting myself on track, for now, CompletionKind should be changed to Keyword and CompletionItemKind to Snippet for self family right?

Reply:
No, I guess `CompletionKind::Magic` is right, so you can leave it like that.

Reply:
Actually `Binding` seems fine since its not a snippet, for `CompletionItemKind`. I'd say keep them as is, `magic` fits better than `keyword` as well as those completions aren't single keyword completions(except for `self`).